### PR TITLE
fix: Improve domain-info validation

### PR DESCRIPTION
**Pull request by rogaldh**
## Description

Validate domain input in the domain-info API route before calling `resolveDomain`. Rejects malformed input.

## Type of change

-   [x] Bug fix

## Screenshots

n/a

## Testing

- Added unit tests for `Domain` superstruct (valid/invalid inputs, including edge cases)

## Related Issues

[HOO-335](https://linear.app/solana-fndn/issue/HOO-335)

## Checklist

-   [x] My code follows the project's style guidelines
-   [x] I have added tests that prove my fix/feature works
-   [x] All tests pass locally and in CI
-   [ ] I have run `build:info` script to update build information
-   [ ] CI/CD checks pass

### CODE REVIEW

File: app/entities/domain/lib/domain-struct.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { define } from 'superstruct';
+
+const hasValidLabels = (hostname: string) =>
+    hostname.includes('.') && !hostname.startsWith('.') && !hostname.endsWith('.') && !hostname.includes('..');
+
+export const Domain = define<string>('Domain', (value) => {
+    if (typeof value !== 'string') return false;
+    try {
+        const { hostname } = new URL(`http://${value}`);
+        return hostname === value.toLowerCase() && hasValidLabels(hostname);
```
[the remark this answers is not in the document]
Comment (rogaldh) [reply app/entities/domain/lib/domain-struct.ts:17]: fixed

File: app/api/domain-info/[domain]/__tests__/route.spec.ts [hunk 2]
```diff
@@ -50,30 +51,87 @@ describe('GET /api/domain-info/[domain]', () => {
...
+        it('should return 400 for input with spaces', async () => {
+            const response = await GET(mockRequest, { params: { domain: 'has spaces.sol' } });
+
+            expect(response.status).toBe(400);
+        });
+
+        it('should return 400 for the URL-encoded failing input from the error log', async () => {
+            const response = await GET(mockRequest, {
+                params: {
+                    domain: 'You%20sent%2075.00%20USDC%20via%20Solana%20network%20To%3A%20BPTAmSr68QhspEN2i8KBKDFjDWbtfAhjiAqU6Cd8H2Yi',
```
[the remark this answers is not in the document]
Comment (rogaldh) [reply app/api/domain-info/[domain]/__tests__/route.spec.ts]: fixed
